### ISSUE TITLE
Unable to reproduce the result achieved on matterport Mask-RCNN with the same config on mmdet

### ISSUE THREAD

**Issue by colt18** [open-mmlab/mmdetection#3946]
This may be seem like a hyperparameter search or find your own cure kinda question but here it goes. 
I do have a dataset consisting of nearly 400 1000*1000px gray level images. There are 3 classes, and annotated objects are no larger than 40x40 pixels. I first give it a go at matterport-mask-rcnn implementation. Training was slow but I was able to achieve approx 0.36 mAP score. After figuring out to implement the same architecture on mmdet, I could only get 0.07 mAP which means there is something going really wrong. For sanity check, json file annotations are showing correctly on builtin viewer function (forgot its name:)). Here is the base config file and things I changed are commented:

```
# model settings
model = dict(
    type='MaskRCNN',
    pretrained=None,
    backbone=dict(
        type='ResNet',
        depth=101,
        num_stages=4,
        out_indices=(0, 1, 2, 3),
        frozen_stages=0, # for retraining all layers
        norm_cfg=dict(type='BN', requires_grad=True),
        norm_eval=False, # Set this to false since pretrained weights are not used (Correct me if I'm wrong)
        style='pytorch'),
    neck=dict(
        type='FPN',
        in_channels=[256, 512, 1024, 2048],
        out_channels=256,
        num_outs=5),
    rpn_head=dict(
        type='RPNHead',
        in_channels=256,
        feat_channels=256,
        anchor_generator=dict(
            type='AnchorGenerator',
            scales=[8],
            ratios=[0.5, 1.0, 2.0],
            strides=[4, 8, 16, 32, 64]),
        bbox_coder=dict(
            type='DeltaXYWHBBoxCoder',
            target_means=[.0, .0, .0, .0],
            target_stds=[1.0, 1.0, 1.0, 1.0]),
        loss_cls=dict(
            type='CrossEntropyLoss', use_sigmoid=True, loss_weight=1.0),
        loss_bbox=dict(type='L1Loss', loss_weight=1.0)),
    roi_head=dict(
        type='StandardRoIHead',
        bbox_roi_extractor=dict(
            type='SingleRoIExtractor',
            roi_layer=dict(type='RoIAlign', output_size=7, sampling_ratio=0),
            out_channels=256,
            featmap_strides=[4, 8, 16, 32]),
        bbox_head=dict(
            type='Shared2FCBBoxHead',
            in_channels=256,
            fc_out_channels=1024,
            roi_feat_size=7,
            num_classes=3, # set classes
            bbox_coder=dict(
                type='DeltaXYWHBBoxCoder',
                target_means=[0., 0., 0., 0.],
                target_stds=[0.1, 0.1, 0.2, 0.2]),
            reg_class_agnostic=False,
            loss_cls=dict(
                type='CrossEntropyLoss', use_sigmoid=False, loss_weight=1.0),
            loss_bbox=dict(type='L1Loss', loss_weight=1.0)),
        mask_roi_extractor=dict(
            type='SingleRoIExtractor',
            roi_layer=dict(type='RoIAlign', output_size=14, sampling_ratio=0),
            out_channels=256,
            featmap_strides=[4, 8, 16, 32]),
        mask_head=dict(
            type='FCNMaskHead',
            num_convs=4,
            in_channels=256,
            conv_out_channels=256,
            num_classes=3, # set classes again
            loss_mask=dict(
                type='CrossEntropyLoss', use_mask=True, loss_weight=1.0))))
# model training and testing settings
train_cfg = dict(
    rpn=dict(
        assigner=dict(
            type='MaxIoUAssigner',
            pos_iou_thr=0.7,
            neg_iou_thr=0.3,
            min_pos_iou=0.3,
            match_low_quality=True,
            ignore_iof_thr=-1),
        sampler=dict(
            type='RandomSampler',
            num=256,
            pos_fraction=0.5,
            neg_pos_ub=-1,
            add_gt_as_proposals=False),
        allowed_border=-1,
        pos_weight=-1,
        debug=False),
    rpn_proposal=dict(
        nms_across_levels=False,
        nms_pre=2000,
        nms_post=1000,
        max_num=1000,
        nms_thr=0.7,
        min_bbox_size=0),
    rcnn=dict(
        assigner=dict(
            type='MaxIoUAssigner',
            pos_iou_thr=0.5,
            neg_iou_thr=0.5,
            min_pos_iou=0.5,
            match_low_quality=True,
            ignore_iof_thr=-1),
        sampler=dict(
            type='RandomSampler',
            num=512,
            pos_fraction=0.25,
            neg_pos_ub=-1,
            add_gt_as_proposals=True),
        mask_size=28,
        pos_weight=-1,
        debug=False))
test_cfg = dict(
    rpn=dict(
        nms_across_levels=False,
        nms_pre=1000,
        nms_post=1000,
        max_num=1000,
        nms_thr=0.7,
        min_bbox_size=0),
    rcnn=dict(
        score_thr=0.05,
        nms=dict(type='nms', iou_threshold=0.5),
        max_per_img=100,
        mask_thr_binary=0.5))

#Dataset
dataset_type = 'CocoDataset'
img_norm_cfg = dict(
    mean=[123.675, 116.28, 103.53], std=[58.395, 57.12, 57.375], to_rgb=False) # Set it false since I'm grayscale (Again correct me if is is not proper)
train_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='LoadAnnotations', with_bbox=True, with_mask=True),
    dict(type='Resize', img_scale=(1000, 1000), keep_ratio=True),
    dict(type='RandomFlip', flip_ratio=0.0), # disabled flip for reproducbility. Also main.py file has some seeds. 
    dict(type='Normalize', **img_norm_cfg),
    dict(type='Pad', size_divisor=32),
    dict(type='DefaultFormatBundle'),
    dict(type='Collect', keys=['img', 'gt_bboxes', 'gt_labels', 'gt_masks']),
]
test_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(
        type='MultiScaleFlipAug',
        img_scale=(1000, 1000),
        flip=False,
        transforms=[
            dict(type='Resize', keep_ratio=True),
            dict(type='RandomFlip'),
            dict(type='Normalize', **img_norm_cfg),
            dict(type='Pad', size_divisor=32),
            dict(type='ImageToTensor', keys=['img']),
            dict(type='Collect', keys=['img']),
        ])
]

classes = ('a', 'b', 'c')
data = dict(
    samples_per_gpu=1,
    workers_per_gpu=2,
    train=dict(
        type=dataset_type,
        ann_file='pathtotrainjson',
        img_prefix='pathtotrainds',
        classes=classes,
        pipeline = train_pipeline),
    val=dict(
        type=dataset_type,
        ann_file='pathtovaljson',
        img_prefix='pathtovalidds',
        classes=classes,
        pipeline = train_pipeline),
    test=dict(
        type=dataset_type,
        img_prefix='pathtotestds',
        pipeline=test_pipeline)) # I don't use validation. I do just train till it converges. 
evaluation = dict(metric=['bbox', 'segm'])

#schedule
optimizer = dict(type='SGD', lr=0.02 / 8, momentum=0.9, weight_decay=0.0001)
optimizer_config = dict(grad_clip=None)
lr_config = dict(
policy='step',
warmup='linear',
warmup_iters=500,
warmup_ratio=0.001,
step=[8, 11])
total_epochs = 100

#default runtime
checkpoint_config = dict(interval=100)
log_config = dict(interval=10, hooks=[dict(type='TextLoggerHook')])
dist_params = dict(backend='nccl')
log_level = 'INFO'
load_from = None
resume_from = None
workflow = [('train', 1)]
work_dir = './work_dirs/blabla'
gpu_ids = range(0, 1)

```

And here is after 30th epoch:
`{"mode": "train", "epoch": 30, "iter": 390, "lr": 3e-05, "memory": 4216, "data_time": 0.03175, "loss_rpn_cls": 0.0102, "loss_rpn_bbox": 0.01336, "loss_cls": 0.04268, "acc": 98.45703, "loss_bbox": 0.08635, "loss_mask": 0.20926, "loss": 0.36185, "time": 0.51338}`

**Comment by v-qjqs** [comment]
Hi @colt18, I'm sorry I might not give you some useful suggestions yet. Do you find something wrong yet?

**Comment by colt18** [comment]
@v-qjqs Well it seems that default config for mrcnn doesn't works for my dataset even I tried out many parameters and it didn't go well. So I just gave up trying. For now, matterport mrcnn works better for my case and looks like I'll stick to that even though I'd like to work with mmdet. Well, obviously this thing is a parameter search and I'm out of luck for sure.
